examples/feature_demo/gltf_viewer.py

```diff
@@ -0,0 +1,274 @@
+                    model_folder
+                    + f"/{model_desc["name"]}/{variant}/{model_desc["variants"][variant]}"
+                )
+                print("Loading model", model_path)
+
+                state["loading"] = True
+                load_model(model_path)
+                state["loading"] = False
+                return
+
+    print("Model not found", model_desc)
+
+
+open_file_dialog = None
+
+
+def load_model(model_path):
+    global model_obj, skeleton_helper, actions
+    try:
+        gltf = asyncio.run(gfx.load_gltf_async(model_path))
+
+        if model_obj:
+            scene.remove(model_obj)
+        if skeleton_helper:
+            scene.remove(skeleton_helper)
+
+        model_obj = gltf.scene if gltf.scene else gltf.scenes[0]
+        model_obj.traverse(add_env_map)
+
+        skeleton_helper = gfx.SkeletonHelper(model_obj)
+        scene.add(skeleton_helper)
+        scene.add(model_obj)
+        state["selected_action"] = 0
+
+        camera.show_object(model_obj, scale=1.4)
+        actions = None
+        clips = gltf.animations
+        if clips:
+            actions = [mixer.clip_action(clip) for clip in clips]
+            if state["animate"]:
+                actions[state["selected_action"]].play()
+    except Exception as e:
+        print(e)
+
+
+def draw_imgui():
+    global model_obj, skeleton_helper, actions, open_file_dialog
+    imgui.new_frame()
+    imgui.set_next_window_size((250, 0), imgui.Cond_.always)
+    imgui.set_next_window_pos(
+        (gui_renderer.backend.io.display_size.x - 250, 0), imgui.Cond_.always
+    )
+    is_expand, _ = imgui.begin(
+        "Controls",
+        None,
+        flags=imgui.WindowFlags_.no_move | imgui.WindowFlags_.no_resize,
+    )
+    if is_expand:
+        if imgui.collapsing_header("Models", imgui.TreeNodeFlags_.default_open):
+            imgui.begin_disabled(state["loading"])
+
+            selected, state["selected_model"] = imgui.combo(
+                " ",
+                state["selected_model"],
+                [m["name"] for m in model_list],
+                10,
+            )
+
+            if selected:
+                threading.Thread(
+                    target=load_remote_model, args=(state["selected_model"],)
+                ).start()
+
+            imgui.end_disabled()
+            if state["loading"]:
+                imgui.same_line()
+                imspinner.spinner_arc_rotation(
+                    "loading", 8, 4.0, imgui.ImColor(0.3, 0.5, 0.9, 1.0), speed=1.0
+                )
+
+            if imgui.button("Open local model file"):
+                open_file_dialog = pfd.open_file(
+                    "Select file", ".", filters=["*.gltf, *.glb", "*.gltf *.glb"]
+                )
+
+            if open_file_dialog is not None and open_file_dialog.ready():
+                files = open_file_dialog.result()
+                if files:
+                    threading.Thread(target=load_model, args=(files[0],)).start()
```

Comment:
Hehe, smart. This way it works whether running on asyncio or qt or anything else.